### PR TITLE
add act api formatters

### PR DESCRIPTION
**Pull request by nathanchica**
This pull request adds comprehensive unit tests for the formatter functions that process bus positions, bus stop profiles, and bus stop predictions from both ACT Realtime and GTFS feeds. The tests cover a wide range of scenarios, including valid and invalid data, edge cases, and fallback logic, ensuring robust and predictable behavior of the formatter functions.

### Bus Position Formatter Tests

* Added tests for `createBusPositionsFromActRealtime` and `createBusPositionsFromGtfsFeed`, verifying parsing, conversion of units, tripId resolution, sorting, filtering of invalid entries, and handling of edge cases like missing or non-finite values.

### Bus Stop Profile Formatter Tests

* Added tests for `createBusStopProfile`, ensuring correct mapping from raw data to structured profiles and enforcing required fields (`geoid` and `stpid`), with error handling for missing data.

### Bus Stop Prediction Formatter Tests

* Added tests for `createBusStopPredictionsFromActRealtime` and `createBusStopPredictionsFromGtfsFeed`, covering filtering by outbound/inbound, required field enforcement, sorting, handling of time calculations, fallback logic for missing or invalid fields, and clamping of negative/invalid values.

### CODE REVIEW

File: src/formatters/busPosition.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    const positions = feedMessage.entity.flatMap((entity) => {
+        const vehicle = entity.vehicle;
+        if (!vehicle) {
+            return [];
+        }
+
+        const routeId = vehicle.trip?.routeId;
+        const latitude = parseFiniteNumber(vehicle.position?.latitude);
+        const longitude = parseFiniteNumber(vehicle.position?.longitude);
+        const vehicleId = vehicle.vehicle?.id ?? entity.id /* v8 ignore next */ ?? '';
```
[the remark this answers is not in the document]
Comment (nathanchica) [reply src/formatters/busPosition.ts:64]: i'm ignoring coverage for the fallback since it's not practically reachable and not worth testing
